src/hooks/useScrollToSlug.ts

```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+import {useParams} from 'react-router-dom';
+
+/**
+ * Reads the slug of a route and scrolls to an element with this id,
+ * if available on the page.
+ *
+ * @see https://github.com/rafgraph/react-router-hash-link/issues/13
+ */
+export function useScrollToSlug() {
+  const {slug} = useParams();
```

Comment:
I think there is an interface missing, see(https://stackoverflow.com/a/60080307)
```suggestion
interface IParamTypes {
 slug: string
}
  const {slug} = useParams<IParamTypes>();
```

Reply:
Thanks, I've added the typings.